jobs/haproxy/spec

```diff
@@ -369,6 +369,12 @@ properties:
   ha_proxy.dns_hold:
     description: "DNS Hold time"
     default: 10s
+  ha_proxy.timeout_retry:
```

Comment:
That property name seems a bit too generic. It should be clear that this timeout belongs to DNS resolution.
Following the pattern from other timeouts I suggest `resolve_retry_timeout`